src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1071,38 +1034,162 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
             cleanUpBuildings(event);
 
             //  Spawn Citizens
-            if (townHall != null && citizens.size() < maxCitizens)
-            {
+            if (townHall != null && citizens.size() < maxCitizens) {
                 int respawnInterval = Configurations.citizenRespawnInterval * 20;
                 respawnInterval -= (60 * townHall.getBuildingLevel());
 
-                if (event.world.getWorldTime() % respawnInterval == 0)
-                {
+                if (event.world.getWorldTime() % respawnInterval == 0) {
                     spawnCitizen();
                 }
             }
+
+            if (!RAID_HAS_HAPPENED && !subscribers.isEmpty() && world != null && !world.isDaytime() && Configurations.doBarbariansSpawn)
```

Comment:
yep, i can do that. the raidHasHappened got changed to lowercase (codify pointed it out, i see what it's used for now)